src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -0,0 +1,932 @@
+    devices += media_user_dir.entryInfoList(
+        QDir::AllDirs | QDir::NoDotAndDotDot);
+
+    // Add folders under /run/media/$USER to devices.
+    QDir run_media_user_dir("/run/media/" + qgetenv("USER"));
+    devices += run_media_user_dir.entryInfoList(
+        QDir::AllDirs | QDir::NoDotAndDotDot);
+
+    foreach(QFileInfo device, devices) {    
+         QFileInfo rbDBFileInfo(device.filePath() + "/" + PDB_PATH);
+
+        if (rbDBFileInfo.exists() && rbDBFileInfo.isFile()) {
+            TreeItem *foundDevice = new TreeItem(this);
+            QList<QString> data;
+
+            data << device.filePath();
+            data << IS_RECORDBOX_DEVICE;
+
+            foundDevice->setLabel(device.fileName());
+            foundDevice->setData(QVariant(data));
+
+            foundDevices << foundDevice;                     
+        }       
+    }
+#else // __APPLE__
+    QFileInfoList devices = QDir("/Volumes").entryInfoList(
+        QDir::AllDirs | QDir::NoDotAndDotDot);
+
+    foreach(QFileInfo device, devices) {  
+         QFileInfo rbDBFileInfo(device.filePath() + "/" + PDB_PATH);
+
+        if (rbDBFileInfo.exists() && rbDBFileInfo.isFile()) {
+            TreeItem *foundDevice = new TreeItem(this);
+            QList<QString> data;
+
+            data << device.filePath();
+            data << IS_RECORDBOX_DEVICE;
+
+            foundDevice->setLabel(device.fileName());
+            foundDevice->setData(QVariant(data));
+
+            foundDevices << foundDevice;               
+        }   
+    }    
+#endif
+
+     return foundDevices;
+}
+
+// Functions getText and parseDeviceDB are roughly based on the following Java file:
+// https://github.com/Deep-Symmetry/crate-digger/blob/master/src/main/java/org/deepsymmetry/cratedigger/Database.java
+// getText is needed because the strings in the PDB file "have a variety of obscure representations".
+
+std::string RekordboxFeature::getText(rekordbox_pdb_t::device_sql_string_t *deviceString) {
+    if (instanceof<rekordbox_pdb_t::device_sql_short_ascii_t>(deviceString->body())) {
+        rekordbox_pdb_t::device_sql_short_ascii_t *shortAsciiString = 
+            static_cast<rekordbox_pdb_t::device_sql_short_ascii_t*>(deviceString->body());
+        return shortAsciiString->text();
+    } else if (instanceof<rekordbox_pdb_t::device_sql_long_ascii_t>(deviceString->body())) {
+        rekordbox_pdb_t::device_sql_long_ascii_t *longAsciiString = 
+            static_cast<rekordbox_pdb_t::device_sql_long_ascii_t*>(deviceString->body());
+        return longAsciiString->text();
+    } else if (instanceof<rekordbox_pdb_t::device_sql_long_utf16be_t>(deviceString->body())) {
+        rekordbox_pdb_t::device_sql_long_utf16be_t *longUtf16beString = 
+            static_cast<rekordbox_pdb_t::device_sql_long_utf16be_t*>(deviceString->body());
+        return longUtf16beString->text();
+    }
+
+    return std::string("");
+}
+
+QString RekordboxFeature::parseDeviceDB(TreeItem *deviceItem) {
```

Comment:
This complex function should be decomposed into smaller chunks of code. Try to extract some functions.